.github/workflows/build-firmware-and-emulator.yml

```diff
@@ -67,13 +67,24 @@ jobs:
     #### Platform-specific Dependencies
     - name: Install msys64 packages
       if: matrix.family == 'windows'
-      run: |
-        C:\msys64\usr\bin\bash -lc 'pacman --noconfirm -S base-devel gcc gdb zip mingw-w64-x86_64-graphviz mingw-w64-x86_64-cppcheck doxygen'
+      uses: msys2/setup-msys2@v2
```

Comment:
Seems to be failing the windows build at the moment, maybe something to do with this change?
```
Run $env:path = $env:path.Insert($env:path.ToLower().IndexOf("c:\windows\system32"), "C:\msys64\mingw64\bin;C:\msys64\usr\bin;")
Setting IDF_PATH: D:\a\_temp\esp-idf
Checking Python compatibility
Adding ESP-IDF tools to PATH...
ERROR: MSys/Mingw is not supported. Please follow the getting started guide of the documentation to set up a supported environment
idf.py: D:\a\_temp\8881f1a1-29ef-4568-9cd8-c3b53623f25f.ps1:4
Line |
   4 |  idf.py build dfu
     |  ~~~~~~
     | The term 'idf.py' is not recognized as a name of a cmdlet, function, script file, or executable program. Check
     | the spelling of the name, or if a path was included, verify that the path is correct and try again.
Error: Process completed with exit code 1.
```

Reply:
Damn, it seemed so close! I'll have to try this on a different PR, I really want to be able to cache msys but it's doing something a little weird and doesn't seem to respect the install path completely.

Reply:
You should only need msys for the emulator step, not the firmware step. The IDF should cover all of that. Maybe try flipping them, building the firmware, then setting up msys2, then building the emulator?

Reply:
Well, reverting it worked too, so that's cool!